drivers/i2c/Kconfig.esp32

```diff
@@ -0,0 +1,102 @@
+# Kconfig.esp32 - ESP32 I2C configuration options
+#
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config I2C_ESP32
+	bool "ESP32 I2C"
+	depends on I2C && SOC_ESP32 && GPIO_ESP32
+	default n
+	help
+	  Enables the ESP32 I2C driver
+
+if I2C_ESP32
+
+config I2C_ESP32_TIMEOUT
+	int "I2C timeout to receive a data bit in APB clock cycles"
+	default 200000
+
+if I2C_0
+
+config I2C_0_DEFAULT_CFG
+	# Standard speed, master
+	default 0x12
+
+config I2C_ESP32_0_TX_LSB_FIRST
+	bool "Port 0 Transmit LSB first"
+	default n
+
+config I2C_ESP32_0_RX_LSB_FIRST
+	bool "Port 0 Receive LSB first"
+	default n
+
+config I2C_ESP32_0_IRQ
+	int "Port 0 IRQ line"
+	default 8
+
+config I2C_ESP32_0_SCL_PIN
+	int "Port 0 SCL pin"
+	default 2
```

Comment:
Likewise, unless the internal mux allows for moving the pins around this seems like something that shouldn't be in kconfig.

Reply:
Any GPIO pin can be SCL or SDA.